### PR TITLE
PE-104 | Improved navigation routes. Also: Cleaned code, Fixed bugs on Profile and More Details pages

### PR DESCRIPTION
**Pull request by MaxLian11**
What I Did:
- Updated ratings_sceen.dart file
- Now after closing the question (rating the user or skipping rating), the question owner will be navigated to the updated question details page. Same after answering
- Fixed errors on more details page
- Added drop-down menu on More Details page for question owner. It includes Close, Answer, and Archive question selections.
- Added changes to edit profile screen: it won't remove the old images if one wasn't selected (bug fixed). Navigation was also changed to "pushAndRemoveUntil" so that we can see the new changes.
- Fixed bestAnswer in the list of answers on Profile Screen

How to Test:
- User must be navigated to question details after closing the question or answering any question, and the back button must bring to a home page
- There must be no errors on any of the pages
- User must navigate back after editing the profile and see the new changes appearing in the profile. If the image wasn't added, it won't remove it like previously.
- Archived questions in the profile must have "archived" status
- Best answers must have the best answer status
- Dropdown menu must be visible only to the question owner, otherwise, it will show a chat icon
- Dropdown selection must vary based on different conditions: Close selection will be visible if question is waiting for answers; Answer selection will be visible if the question is not closed yet. The archive selection will be visible always. If the question is archived, the dropdown menu won't show up at all.

**All edits are open to suggestions**

### CODE REVIEW

**Comment by tonynguyen98** [comment]
The things you've fixed is working as intended! However, I've found another bug that can be fixed in your branch. The archive button overflows the Category: 
Screenshot_20210502-002330

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
Everything is working as intended, but small UI issue as stated above.

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
I'm not sure how I feel about moving the `Close Question` into the drop down. I would prefer that it was still a button on the bottom, but I'm okay with answer and archive being in there though.

File: lib/components/eureka_rounded_button.dart [hunk 1]
```diff
@@ -8,14 +8,16 @@ class EurekaRoundedButton extends StatefulWidget {
   final String buttonText;
   final Color buttonColor;
   final Color textColor;
+  final bool isTwoButtons;
 
   /// Users will need to pass in values for the onPressed interation and
   /// what the text for this button will be.
```
Comment (txt020) [inline lib/components/eureka_rounded_button.dart:14]: Remove comments if they are not necessary.

**Review by txt020 — APPROVED** [review]

**Comment by txt020** [comment]
Everything works accordingly. Great Job!

File: lib/components/eureka_rounded_button.dart (hunk 1 above)
Comment (MaxLian11) [reply lib/components/eureka_rounded_button.dart:14]: I think these kinds of comments useful

Comment (txt020) [reply lib/components/eureka_rounded_button.dart:14]: For sure.

**Review by tonynguyen98 — APPROVED** [review]
Great work! I like it now!

**Pull request merged**